lib/elixir/lib/string_io.ex

```diff
@@ -23,7 +23,58 @@ defmodule StringIO do
 
   If the `:capture_prompt` option is set to `true`,
   prompts (specified as arguments to `IO.get*` functions)
-  are captured.
+  are captured in the output.
+
+  The device will be created and sent to the function given.
+  When the function returns, the device will be closed. The final
+  result will be a tuple with `:ok` and the result of the function.
+
+  ## Examples
+      iex> {:ok, _contents} = StringIO.open("foo", [], fn(pid) ->
+      ...>     input = IO.gets(pid, ">")
+      ...>     IO.write(pid, "The input was \#{input}")
+      ...>     StringIO.contents(pid)
+      ...>   end)
+      {:ok, {"", "The input was foo"}}
+
+      iex> {:ok, _contents} = StringIO.open("foo", [capture_prompt: true], fn(pid) ->
+      ...>     input = IO.gets(pid, ">")
+      ...>     IO.write(pid, "The input was \#{input}")
+      ...>     StringIO.contents(pid)
+      ...>   end)
+      {:ok, {"", ">The input was foo"}}
+  """
+  @spec open(binary, keyword, (pid -> res)) :: {:ok, res} when res: var
+  def open(string, options, function)
+      when is_binary(string) and is_list(options) and is_function(function, 1) do
+    with {:ok, pid} <- GenServer.start_link(__MODULE__, {string, options}, []) do
```

Comment:
Given it is a single clause, I think `case` would be clearer here.

Reply:
OK I think I see what you mean.  I will change the "with" to a case